src/ORM/Queries/SQLSelect.php

```diff
@@ -796,4 +820,49 @@ public function isEmpty()
         // Empty if there's no select, or we're trying to select '*' but there's no FROM clause
         return empty($this->select) || (empty($this->from) && array_key_exists('*', $this->select));
     }
+
+    public function setUseCache(bool $useCache, string $namespace = 'default'): static
+    {
+        $this->useCache = $useCache;
+        $this->cacheNamespace = $namespace;
+        return $this;
+    }
```

Comment:
The namespace here is important - it's what allows us to invalidate a cache for one `DataObject` class without affecting others.
It could also be useful for projects to give their own custom queries a namespace, but that's just a side effect.